environments/linux/docker/icc.base.Dockerfile

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2018-2020 NVIDIA Corporation
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Released under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+
+ARG ROOT_IMAGE
+
+FROM ${ROOT_IMAGE} AS devenv
+
+ARG COMPILERS="gcc clang"
+ARG ICC_TOOLKIT_VER="latest"
+ARG CMAKE_VER=3.23.1
+ARG CMAKE_URL=https://github.com/Kitware/CMake/releases/download/v${CMAKE_VER}/cmake-${CMAKE_VER}-Linux-x86_64.sh
+
+# `-y` answers yes to any interactive prompts.
+# `-qq` because apt is noisy
+ARG APT_GET="apt-get -y -qq"
+
+ENV TZ=US/Pacific
+ENV DEBIAN_FRONTEND=noninteractive
+# apt-key complains about non-interactive usage.
+ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1
+
+SHELL ["/usr/bin/env", "bash", "-c"]
+
+ADD ${CMAKE_URL} /tmp/cmake.sh
+
+# Install baseline development tools
+RUN function comment() { :; }; \
+    comment "Sources for gcc"; \
+    source /etc/os-release; \
+    ${APT_GET} update; \
+    comment "Install basic build tools"; \
+    ${APT_GET} --no-install-recommends install apt-utils curl wget git zip unzip tar \
+        sudo make software-properties-common ninja-build ccache pkg-config \
+        python3 python3-wheel python3-pip; \
+    comment "Install GCC and Clang"; \
+    # Unattended installation hack
+    ${APT_GET} install gcc g++ ${COMPILERS}; \
+    comment "Install CMake"; \
+    sh /tmp/cmake.sh --skip-license --prefix=/usr; \
+    comment "Install Python utils"; \
+    update-alternatives --quiet --install /usr/bin/python python $(which python3) 3; \
+    update-alternatives --quiet --set python $(which python3); \
```

Comment:
We can install `python-is-python3` :)

Reply:
I would not want to do any changes in a bugfix PR. I guess we all agree that our CI images need some love but that here is not the right PR for it

Reply:
Yeah, that's something we can fix later.